### PR TITLE
Rename GND_* parameters to BARO*

### PR DESCRIPTION
**Pull request by tridge**
This gives us the following parameters for 3 baros:
```
BARO1_DEVID      65540
BARO1_GND_PRESS  94504
BARO2_DEVID      0.000000
BARO2_GND_PRESS  94503
BARO3_DEVID      0.000000
BARO3_GND_PRESS  94502
BARO_GND_TEMP    0.000000
BARO_ALT_OFFSET  0.000000
BARO_EXT_BUS     -1.000000
BARO_FLTR_RNG    0.000000
BARO_PRIMARY     0.000000
BARO_PROBE_EXT   0.000000
```
It marks the pressure and temperature as internal only which prevents users writing to the params. It also adds a new BRD_OPTIONS bit which allows for write to internal params. This will give us an easy way to handle parameters that should normally not be written, but which developers may sometimes want to set

### CODE REVIEW

File: Tools/Frame_params/Sub/bluerov2-heavy-3_5_2.params [hunk 1]
```diff
@@ -216,11 +216,6 @@ FS_TEMP_ENABLE,0
 FS_TEMP_MAX,62
 FS_TERRAIN_ENAB,0
 GCS_PID_MASK,0
-GND_ALT_OFFSET,0.000000000000000000
-GND_EXT_BUS,1
```
Comment (amilcarlucas) [inline Tools/Frame_params/Sub/bluerov2-heavy-3_5_2.params:220]: GND_EXT_BUS is 1 here, not -1. Shoun't that be preserved?

File: Tools/Frame_params/Sub/bluerov2-3_5_2.params [hunk 2]
```diff
@@ -215,11 +215,6 @@ FS_TEMP_ENABLE,0
 FS_TEMP_MAX,62
 FS_TERRAIN_ENAB,0
 GCS_PID_MASK,0
-GND_ALT_OFFSET,0.0
-GND_EXT_BUS,1
```
Comment (amilcarlucas) [inline Tools/Frame_params/Sub/bluerov2-3_5_2.params:219]: GND_EXT_BUS is 1 here, not -1. Shoun't that be preserved?

File: Tools/Frame_params/Sub/bluerov2-3_5.params [hunk 3]
```diff
@@ -218,11 +218,6 @@ FS_TEMP_ENABLE,0
 FS_TEMP_MAX,62
 FS_TERRAIN_ENAB,0
 GCS_PID_MASK,0
-GND_ALT_OFFSET,0.000000000000000000
-GND_EXT_BUS,1
```
Comment (amilcarlucas) [inline Tools/Frame_params/Sub/bluerov2-3_5.params:222]: GND_EXT_BUS is 1 here, not -1. Shoun't that be preserved?

File: libraries/AP_Baro/AP_Baro.cpp [hunk 4]
```diff
@@ -77,125 +77,125 @@ const AP_Param::GroupInfo AP_Baro::var_info[] = {
...
     // @Units: Pa
     // @Increment: 1
     // @ReadOnly: True
     // @Volatile: True
     // @User: Advanced
-    AP_GROUPINFO("ABS_PRESS", 2, AP_Baro, sensors[0].ground_pressure, 0),
+    AP_GROUPINFO_FLAGS("1_GND_PRESS", 2, AP_Baro, sensors[0].ground_pressure, 0, AP_PARAM_FLAG_INTERNAL_USE_ONLY),
 
-    // @Param: TEMP
+    // @Param: _TEMP
```
Comment (amilcarlucas) [inline libraries/AP_Baro/AP_Baro.cpp:90]: should this be 1_TEMP, or _TEMP ?

File: libraries/AP_Baro/AP_Baro.cpp [hunk 5]
```diff
@@ -77,125 +77,125 @@ const AP_Param::GroupInfo AP_Baro::var_info[] = {
...
     // @DisplayName: ground temperature
     // @Description: User provided ambient ground temperature in degrees Celsius. This is used to improve the calculation of the altitude the vehicle is at. This parameter is not persistent and will be reset to 0 every time the vehicle is rebooted. A value of 0 means use the internal measurement ambient temperature.
     // @Units: degC
     // @Increment: 1
     // @Volatile: True
     // @User: Advanced
-    AP_GROUPINFO("TEMP", 3, AP_Baro, _user_ground_temperature, 0),
+    AP_GROUPINFO_FLAGS("1_TEMP", 3, AP_Baro, _user_ground_temperature, 0, AP_PARAM_FLAG_INTERNAL_USE_ONLY),
```
Comment (amilcarlucas) [inline libraries/AP_Baro/AP_Baro.cpp:97]: Inconsistent with parameter name above.

**Review by amilcarlucas — CHANGES_REQUESTED** [review]

File: Tools/Frame_params/Sub/bluerov2-heavy-3_5_2.params (hunk 1 above)
Comment (tridge) [reply Tools/Frame_params/Sub/bluerov2-heavy-3_5_2.params:220]: the sub code overrides it

File: Tools/Frame_params/Sub/bluerov2-3_5.params (hunk 3 above)
Comment (tridge) [reply Tools/Frame_params/Sub/bluerov2-3_5.params:222]: sub overrides it

File: libraries/AP_Baro/AP_Baro.cpp (hunk 4 above)
Comment (tridge) [reply libraries/AP_Baro/AP_Baro.cpp]: thanks! fixed

File: libraries/AP_Baro/AP_Baro.cpp (hunk 5 above)
Comment (tridge) [reply libraries/AP_Baro/AP_Baro.cpp:97]: thanks! fixed

File: libraries/AP_Baro/AP_Baro.cpp [hunk 6]
```diff
@@ -77,125 +77,125 @@ const AP_Param::GroupInfo AP_Baro::var_info[] = {
...
     // @Units: Pa
     // @Increment: 1
     // @ReadOnly: True
     // @Volatile: True
     // @User: Advanced
-    AP_GROUPINFO("ABS_PRESS", 2, AP_Baro, sensors[0].ground_pressure, 0),
+    AP_GROUPINFO_FLAGS("1_GND_PRESS", 2, AP_Baro, sensors[0].ground_pressure, 0, AP_PARAM_FLAG_INTERNAL_USE_ONLY),
 
-    // @Param: TEMP
+    // @Param: 1_TEMP
```
Comment (rmackay9) [inline libraries/AP_Baro/AP_Baro.cpp:90]: Don't want to rename it to be BARO1_GND_TEMP while we're modifying it?

**Comment by rmackay9** [comment]
Very happy to see this renaming!

@meee1 does MP manually set any GND_ parameters?

File: libraries/AP_Baro/AP_Baro.cpp (hunk 6 above)
Comment (tridge) [reply libraries/AP_Baro/AP_Baro.cpp]: i've changed it to BARO_GND_TEMP
it is actually not per-baro

File: libraries/AP_Baro/AP_Baro.cpp [hunk 7]
```diff
@@ -77,125 +77,125 @@ const AP_Param::GroupInfo AP_Baro::var_info[] = {
     // NOTE: Index numbers 0 and 1 were for the old integer
     // ground temperature and pressure
 
-    // @Param: ABS_PRESS
-    // @DisplayName: Absolute Pressure
+    // @Param: 1_GND_PRESS
+    // @DisplayName: Ground Pressure
```
Comment (rmackay9) [inline libraries/AP_Baro/AP_Baro.cpp:81]: Ideally we should add "Baro" to the front of the display name.

File: libraries/AP_Baro/AP_Baro.cpp [hunk 8]
```diff
@@ -77,125 +77,125 @@ const AP_Param::GroupInfo AP_Baro::var_info[] = {
...
     // @User: Advanced
-    AP_GROUPINFO("ABS_PRESS3", 11, AP_Baro, sensors[2].ground_pressure, 0),
+    AP_GROUPINFO_FLAGS("3_GND_PRESS", 11, AP_Baro, sensors[2].ground_pressure, 0, AP_PARAM_FLAG_INTERNAL_USE_ONLY),
 
     // Slot 12 used to be TEMP3
 #endif
 
-    // @Param: FLTR_RNG
+    // @Param: _FLTR_RNG
     // @DisplayName: Range in which sample is accepted
```
Comment (rmackay9) [inline libraries/AP_Baro/AP_Baro.cpp:159]: An existing problem but this display name is pretty terrible.  It gives no indication to the user what it's affecting without looking at the actual parameter name.

**Comment by tridge** [comment]
test:
 - QGC
 - MissionPlanner

**Review by peterbarker — APPROVED** [review]
LGTM but needs testing against MissionPlanner (and as a stretch goal QGC)

**Comment by tridge** [comment]
I've tested MissionPlanner and QGC.
 - MissionPlanner: no issues found
 - QGC: only issue was lack of COMPASS_PRIMARY, no complaint on the baro parms

**Pull request merged**